x-pack/platform/plugins/private/upgrade_assistant/server/lib/es_deprecations_status/index.ts

```diff
@@ -56,14 +56,22 @@ export async function getESUpgradeStatus(
       return status !== 'green';
     }) as EnrichedDeprecationInfo[];
 
-    return [...enrichedHealthIndicators, ...toggledMigrationsDeprecations];
+    return {
+      enrichedHealthIndicators,
+      migrationsDeprecations: toggledMigrationsDeprecations,
+    };
   };
+  const { enrichedHealthIndicators, migrationsDeprecations } = await getCombinedDeprecations();
```

Comment:
Keeping the indicators separate allows for blocking minor upgrades on health-only issues and ignoring (and not showing) critical deprecations that only need to be handled when upgrading to the next major.